src/web/lib/braze/checkBrazeDependencies.ts

```diff
@@ -0,0 +1,100 @@
+import { getBrazeUuid } from '@root/src/web/lib/getBrazeUuid';
+import { hasRequiredConsents } from './hasRequiredConsents';
+import { hideSupportMessaging } from './hideSupportMessaging';
+
+type SuccessResult = {
+	isSuccessful: true;
+	data: ResultData;
+};
+
+type FailureResult = {
+	isSuccessful: false;
+	failureField: string;
+	failureData: any;
+	data: ResultData;
+};
+
+type DependenciesResult = SuccessResult | FailureResult;
+
+type ResultData = { [key: string]: string | boolean };
+
+type DependencyResult = {
+	name: string;
+	value: Promise<string | boolean | null | undefined>;
+};
+const buildFailureResponse = (name: string, value: any, data: ResultData) => ({
+	isSuccessful: false,
+	failureField: name,
+	failureData: value,
+	data,
+});
+
+const buildDependencies = (
+	isSignedIn: boolean,
+	idApiUrl: string,
+): DependencyResult[] => {
+	return [
+		{
+			name: 'brazeSwitch',
+			value: Promise.resolve(window.guardian.config.switches.brazeSwitch),
+		},
+		{
+			name: 'apiKey',
+			value: Promise.resolve(window.guardian.config.page.brazeApiKey),
+		},
+		{
+			name: 'brazeUuid',
+			value: isSignedIn ? getBrazeUuid(idApiUrl) : Promise.resolve(null),
+		},
+		{
+			name: 'consent',
+			value: hasRequiredConsents(),
+		},
+		{
+			name: 'userIsGuSupporter',
+			value: Promise.resolve(hideSupportMessaging()),
+		},
+		{
+			name: 'isNotPaidContent',
+			value: Promise.resolve(!window.guardian.config.page.isPaidContent),
+		},
+	];
+};
+
+const checkBrazeDependencies = async (
+	isSignedIn: boolean,
+	idApiUrl: string,
+): Promise<DependenciesResult> => {
+	const dependencies = buildDependencies(isSignedIn, idApiUrl);
+
+	const data: ResultData = {};
+
+	// I think we could possibly clean this up a bit when we can use
+	// Promise.allSettled reliably (it's not available in our current Node
+	// version and polyfill.io doesn't have a polyfill yet).
+	for (const { name, value } of dependencies) {
+		try {
+			// eslint-disable-next-line no-await-in-loop
+			const result = await value;
+
+			if (result) {
+				data[name] = result;
+			} else {
+				return buildFailureResponse(name, result, data);
+			}
```

Comment:
maybe just throw an error here, so `buildFailureResponse` only ever used in catch block?

Reply:
Also if `buildFailureResponse` is only defined once, then maybe we don't need it as a seperate function? we could return the object here?

Reply:
Ah interesting question! This did actually cross my mind, but I decided against (using exceptions for flow control never feels quite right to me). Maybe the scope here is tight enough that it's not a problem though. It would be nice to have a single failure case build in one place only. I'll give it a try and see how it looks!

Reply:
I did try this but it felt clearer to me keeping the falsy case and the catch separate.